Mathlib/Data/Matrix/RowCol.lean

```diff
@@ -27,7 +27,7 @@ variable {R : Type*} {α : Type v} {β : Type w}
 namespace Matrix
 
 /-- `Matrix.col u` is the column matrix whose entries are given by `u`. -/
-def col (w : m → α) : Matrix m Unit α :=
+def col (w : m → α) : Matrix m (Fin 1) α :=
```

Comment:
An alternative would be to have this defined for any `Unique ι`, which may or may not be annoying downstream.

Reply:
I was coming on this PR to say the same thing

Reply:
That sounds like the better idea indeed! Ill make these modifications tomorrow

Reply:
I tried using

```lean
def col {ι : Type} [Unique ι] (w : m → α) : Matrix m ι α :=
  of fun x _ => w x
```

but the implications were that any theorem about `col` and `row` anywhere needed addition of `{ι : Type} [Unique ι]` just to make the statement compile.

Here is a PR fixing just two files with the alternative choice, I think we would see a lot of `(ι := ι)` appear everywhere: #11536

Reply:
Update: #11536 is now also ready for review. The two PRs are mutually exclusive.

I have to admit, I still don't really understand what we gain by generalising which one-element-index-type we are using.

To me it seems that `Fin 1` is only correct choice: For concrete matrices we need `Fin 1` for the notation and if anybody ever defined an abstract matrix on a different type `ι` with `Unique ι`, they could very likely use `Equiv.equivPUnit` to turn that into `Fin 1` instead.

Reply:
Maybe you're right. It depends on how people use it